### ISSUE TITLE
New Dialog Element

### ISSUE THREAD

**Issue by timwright12** [10up/wp-component-library#168]
Considering re-writing modal to use the new dialog element and polyfill: https://keithjgrant.com/posts/2018/meet-the-new-dialog-element/

It has a pretty nice API, doesn't quite pass a11y standards, but we should be able to reuse some of the current JS to bring it up to compliance.

I wouldn't tackle this until we upgrade the components to ES6 though.

**Comment by timwright12** [comment]
Closing this out since we're using the a11y dialog plugin
